### PR TITLE
reduce cost of sorting k8s analysis messages

### LINKED ISSUE #30218: status: excessive CPU on proxy events
Setup:
* Create 3k virtual services
* Spin up 1 pod, then 2, then 3 ..

CPU usage is >1 CPU: 

Without status enabled, CPU usage is ~60m

### PR DESCRIPTION
**Pull request by therealmitchconnors**
fixes #30218

### CODE REVIEW

File: galley/pkg/config/source/kube/rt/origin.go [hunk 1]
```diff
@@ -48,6 +48,10 @@ func (o *Origin) FriendlyName() string {
 	return fmt.Sprintf("%s %s", o.Kind, o.FullName.String())
 }
 
+func (o *Origin) Comparator() string {
+	return o.Kind + o.FullName.Name.String() + o.FullName.Namespace.String()
```
Comment (howardjohn) [inline galley/pkg/config/source/kube/rt/origin.go:52]: nit: this is not unique without delimiters

Comment (howardjohn) [reply galley/pkg/config/source/kube/rt/origin.go:52]: Actually kind is also not distinct (there are two Gateway's). So if we expect them to be unique we will need more info. If its just for sorting its probably less important?

Comment (therealmitchconnors) [reply galley/pkg/config/source/kube/rt/origin.go:52]: Just for sorting.  trying to avoid 3x calls to fmt.Sprintf times n*log(n) for the sort operation.

Comment (jasonwzm) [reply galley/pkg/config/source/kube/rt/origin.go:52]: Having a delimiter would be much better. Kind should be fine, as they are all within Istio resource scope.

Comment (jasonwzm) [reply galley/pkg/config/source/kube/rt/origin.go:52]: @therealmitchconnors can you add a delimiter just to be 100% safe and I will approve. Thanks!

Comment (howardjohn) [reply galley/pkg/config/source/kube/rt/origin.go:52]: ping?

**Review by howardjohn — APPROVED** [review]

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**